# Genome-wide rotational and translational setting of transcription factors with nucleosomes

### Jordan E. Krebs#,1,2, Haining Chen#,2, Olivia W. Lang2, William K. M. Lai2, B. Franklin Pugh2*
### #Co-first author

1MD/PhD Medical Scientist Training Program, Penn State College of Medicine, Hershey, PA, USA.
2Department of Molecular Biology and Genetics, Cornell University, Ithaca, New York, 14853, USA

### Correspondence:fp265@cornell.edu

### PMID : [XXXXXXXX](https://pubmed.ncbi.nlm.nih.gov/XXXXXXXX/)
### GEO ID : [XXXXXXXX](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=XXXXXXXX)

## Abstract
Transcription factors (TFs) regulate gene expression through DNA sequences (TFBS) near target genes. Unbound sites are thought to be generally inaccessible when encased by nucleosomes, the global packaging unit of the genome. Bound sites are associated with nucleosome displacement. Yet the resolution of genome-wide assays has not provided sufficient granularity to show how the DNA helix on nucleosomal surfaces is presented to TFs, and whether TFs continue to engage nucleosomes after nucleosomes have been displaced. Here we use an endonuclease that is sensitive to the rotational setting of nucleosomal DNA to map rotational relationships between TFs, TFBS and nucleosomes within their native genomic chromatin context in human cells. Certain TF motifs are rotationally buried (e.g., ATF) whereas others are rotationally accessible (e.g., USF). Rotational positioning occurs along all 14 helical gyres of nucleosomal DNA. Certain TFs (e.g., CTCF/Cohesin complex and NFIA), when bound to their sites, produce a new rotational and positional relationship with the local nucleosome. When coupled to ChIP-exo, multiple aspects of TF-nucleosome interactions are revealed. This study reveals a consolidated set of themes by which TFs engage nucleosomes in their natural context.


## Table of Contents

### data

### bin

### 00_Download_and_Preprocessing

### 01_Run_GenoPipe

### 02_Call_RefPT

### 0X_Bulk_Processing

### Library

